### ISSUE TITLE
setting VCPKG_PLATFORM_TOOLSET in a custom triplet file has no effect on boost compilation

### ISSUE THREAD

**Issue by eladmaimoni** [microsoft/vcpkg#24373]
**Describe the bug**
I have a simple C++ console application (VS 2022) that uses a custom triplet and a manifest file.
I use the following triplet file **x64-windows-static-md-v143.cmake**:
```
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE static)
set(VCPKG_PLATFORM_TOOLSET v143)
```
I use this simple manifest file vcpkg.json:
```
{
    "name": "em",
    "version": "1.0.0",
    "builtin-baseline" : "1085a57da0725c19e19586025438e8c16f34c890",
    "dependencies": [
        "boost-atomic"
    ]
}
```

I configure Visual Studio to use the triplet, and I also make sure it uses it:
![image](https://user-images.githubusercontent.com/17446835/164968696-99729943-5452-4dce-a83b-cc07d95a2987.png)

After compilation, it seems that the generated boost library is still created with the v140 platform toolset, at least as can be deduced by looking at its name: boost_atomic-**vc140**-mt.lib

It seems that the generated boost library is still created using a different platform toolset - v140 in my case.

**Is this a bug? how can I know for sure which platform toolset boost compilation is using?**

**Environment**
- OS: Windows 10
- Compiler: Visual Studio 2022

**To Reproduce**

1. create an empty Visual Studio 2022 C++ console application
2. configure it to use the provided manifest file and triplet
3. compile and examine the vcpkg_installed directory.

**Comment by Osyotr** [comment]
TLDR; it's ok
See https://github.com/microsoft/vcpkg/issues/601#issuecomment-276220946

**Comment by Adela0814** [comment]
@elad8a Has your problem been solved？

**Comment by elad8a** [comment]
@Adela0814 the code compiles. I don't understand if this is an actual issue other than the apparent name.

**Comment by Adela0814** [comment]
1. Input option `--debug` to `Additional Options`, you will get your current toolset in output box.
![image](https://user-images.githubusercontent.com/95600143/168007266-f99dca02-85e2-4dea-ad8a-d171e698ae43.png)
2. If you want to set a specific toolset, you need to add `set(VCPKG_PLATFORM_TOOLSET v143)` to `vcpkg\triplets\x64-windows.cmake`.

**Comment by Adela0814** [comment]
I'm not sure if you have any other problems with your triplet. For example, `'static'` cannot be included
![image](https://user-images.githubusercontent.com/95600143/168011848-77d2fddf-f225-4de7-b078-07f4403a23bf.png)

You can try the existing triplet first.

**Comment by Adela0814** [comment]
@elad8a ping~

**Comment by Adela0814** [comment]
We hope your question was answered to your satisfaction; if it wasn't, you can reopen with more info.
